debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/transforms/ExtractNewDocumentStateTest.java

```diff
@@ -230,7 +230,9 @@ public void shouldFailWhenTheSchemaLooksValidButDoesNotHaveTheCorrectFields() {
                 valueSchema,
                 value);
 
-        exceptionRule.expect(NullPointerException.class);
+        // Prior to AK 2.4.1, this threw a NullPointerException
+        // As of AK 2.4.1, this now causes an IllegalArgumentException
+        exceptionRule.expect(IllegalArgumentException.class);
```

Comment:
Thinking about it, I suppose this will impact the matrix job for Kafka versions on CI? Might need to have two flavours of this test which are enabled based on the tested Kafka version. WDYT?

Reply:
I agree, but I took a slightly different approach and added a `Testing.Kafka` interface that allows us to test the major/minor/patch version in code and branch accordingly.  If we find we need more of this moving forward we can look into formalizing this into an annotation or something at that point.

Reply:
After discussion with @jpechane we felt it was better to use annotations instead so I added the `SkipWhenKafkaVersion` annotation and its supporting files so we can use this in other spots going forward in tests.

Reply:
Nice!